### PR TITLE
finish

### PR DESCRIPTION
**Pull request by Ramosok**

### CODE REVIEW

File: src/pages/TodoList/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {createAction} from "redux-actions";
+
+export const CREATE_TODO =createAction('CREATE_TODO');
```
Comment (adryianSiarheyeu) [inline src/pages/TodoList/actions/index.js:3]: use prettier

File: src/pages/TodoList/components/TodoEditMode/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import {memo} from "react";
+import {useForm} from "../../../../hooks";
+
+import Card from "@mui/material/Card";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import SaveIcon from '@mui/icons-material/Save';
+import ExitToApp from '@mui/icons-material/ExitToApp';
+import PropTypes from "prop-types";
```
Comment (adryianSiarheyeu) [inline src/pages/TodoList/components/TodoEditMode/index.js:9]: imports could be optimized. 
1. import {Card, TextField, ...etc} from '@mui/material'
2. Sane for Icons

File: src/pages/TodoList/containers/TodoListContainer.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import {TodoLayout} from "../components/TodoLayout";
+
+import {useForm} from "../../../hooks";
+
+import {COMPLETE_TODO, CREATE_TODO, DELETE_TODO, SAVE_TODO, TOGGLE_EDIT_MODE} from "../actions";
+
+export const TodoListContainer = () => {
+    const dispatch = useDispatch();
+
+    const {todos} = useSelector(state => state.todoManagerList)
```
Comment (adryianSiarheyeu) [inline src/pages/TodoList/containers/TodoListContainer.js:13]: use selectors

File: src/pages/TodoList/containers/TodoListContainer.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    const {formValues, handleChange, handleReset} = useForm({taskText: ""});
+
+    const handleTaskCreate = useCallback((event) => {
+        event.preventDefault();
+        if (formValues.taskText) {
+            dispatch(CREATE_TODO(formValues.taskText));
+        }
+        handleReset();
+    }, [dispatch, formValues.taskText, handleReset]);
```
Comment (adryianSiarheyeu) [inline src/pages/TodoList/containers/TodoListContainer.js:23]: why did you add handleReset to depth?

File: src/pages/TodoList/reducers/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                isCompleted: false,
+                taskText: payload,
+                date: date,
+            };
+            return {
+                todos: [newTask, ...state.todos],
+            };
+        },
+        [actions.TOGGLE_EDIT_MODE]: (state, {payload: id}) => {
+            const copyTodos = [...state.todos];
```
Comment (adryianSiarheyeu) [inline src/pages/TodoList/reducers/index.js:27]: use cloneDeep

**Review by adryianSiarheyeu — CHANGES_REQUESTED** [review]

**Comment by Ramosok** [comment]
dsfsdfsd

**Pull request closed without merging**